### ISSUE TITLE
  On the variation page q645, it says t...

### ISSUE THREAD

**Issue by wormbasehelp** [WormBase/website#6362]
*Submitted from the feedback form on the WormBase website.*

<p>&nbsp;&nbsp;On the variation page q645, it says that the substitution is A -> R.  I have recently sequenced this variant as part of a cross, and the sequence I am getting indicates that the substitution is A -> T.</p>

**Reported by:** Jona*********** (jrum**************************)
**Submitted from:** <a target="_blank" href="https://wormbase.org//tools/support">/tools/support</a>
**Browser:** Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:59.0) Gecko/20100101 Firefox/59.0

**Comment by Paul-Davis** [comment]
Dear Jonathan,

Thank you for submitting your observations.

I have just re-read the Siegfried & Kimble, 2002 
<https://wormbase.org/resources/paper/WBPaper00005116> publication and 
they report a Aspartic acid to Glutamic acid peptide change which can 
only be done via a T -> R (A or G) change. Checking the annotation 
everything appears to have been captured correctly. The Molecular 
Details Widget seems to report the reverse strand hence the A -> R. This 
doesn't make sense so I will open a ticket with the web team to look at 
this and get the correct strand reported.  So WT should be T so I'd be 
interested in looking at your sequence data as it seems like a WT sequence?

Paul




On 17/04/2018 21:57, no-reply@wormbase.org wrote:
> Content Correction
>
>   On the variation page q645, it says that the substitution is A -> R. 
> I have recently sequenced this variant as part of a cross, and the 
> sequence I am getting indicates that the substitution is A -> T.
>
> Submitted From:
>   support /tools/support <https://wormbase.org//tools/support>
>
> This issue has been assigned ticket number: #6362. Follow progress on 
> our issue tracker <https://github.com/WormBase/website/issues/6362>.
>
> Reported by: Jonathan Rumley (jrumley@pennmedicine.upenn.edu)
> 17 Apr 18 20:57
>
> Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:59.0) Gecko/20100101 
> Firefox/59.0
>
> This question was reported via the web interface at wormbase.org. You 
> can also email us <mailto:help@wormbase.org> directly.
>
> -- 
> -- 
> You received this message because you are subscribed to the "WormBase: 
> help" group.
> To post to this group, send email to help@wormbase.org
> To unsubscribe from this group, send email to
> help+unsubscribe@wormbase.org
> For more options, visit this group at
> http://groups.google.com/a/wormbase.org/group/help?hl=en
>

-- 
===========================================
WormBase
European Bioinformatics Institute (EMBL-EBI)
European Molecular Biology Laboratory
Wellcome Trust Genome Campus
Hinxton
Cambridge CB10 1SD
United Kingdom
===========================================

**Comment by Paul-Davis** [comment]
Primary annotation:
![var_annot](https://user-images.githubusercontent.com/717707/38935779-1923e030-4317-11e8-8f7b-ed9864655705.png)

Acedb_alignment:
![var_genome](https://user-images.githubusercontent.com/717707/38935798-25256c3c-4317-11e8-8ee6-5045df5e9f0a.png)

Molecular_details:
![var_mol_details](https://user-images.githubusercontent.com/717707/38935818-31946518-4317-11e8-8954-a3b9d5c10f84.png)

**Comment by Paul-Davis** [comment]
No reply from the user as far as I can see.....looks like a WT sequence they have supplied...but the display bug persists as detailed above.

**Comment by Paul-Davis** [comment]
User response:

Dear Paul,

Sorry for my delayed response.  I had not yet had the WT allele from N2 sequenced, and I wanted to include it in my response.  My sequences are from the same strand as is reported on the q645 variation page.  Based on my sequences, it does look like the wild type sequence reported on the q645 page is correct for the strand that is reported.  Based on my mutant sequences, it does not look like R is correct for this strand.  The CDS page for W10C8.2 reports the opposite strand, and has this base as T for WT.  This T is the third base of the codon GAT, which codes for aspartic acid (D).  On this strand, the sequence change I am finding for q645 would be T to A, which would result in the codon GAA, which codes for glutamic acid (E).  Thank you for your time.  Attached are my sequencing results.  I have included in the file names whether I think descendants of my cross have the mutant or WT allele.  The first two files are sequences from N2 worms.  The third file is from dumpy worms of the strain JK2689, which should be homozygous for pop-1(q645). The fourth and fifth files are from descendants of my cross that have the WT sequence.  The rest of the files are from descendants of my cross that have the mutant sequence.

Jonathan

**Comment by Paul-Davis** [comment]
I aligned all there sequence data which nicely agrees with the position and display(ish)

![user_seqs_and_display_seq_alignment](https://user-images.githubusercontent.com/717707/39469353-adb42ec2-4d2f-11e8-8729-f408e763c99f.png)

**Comment by Paul-Davis** [comment]
Hi @sibyl229 thanks for pointing out the note 👍 

It really doesn't make sense for it to be always displayed on the +ve strand as when ambiguity code is used in the annotation the context becomes confused and incorrect. This has confused a user and myself. The CDS summary pages do a better job at relating it to the correct strand and context.

When @mgrbyte looked quickly at the code it seemed that there was a section commented out.

Paul

**Comment by Paul-Davis** [comment]
Another user has reported a problem with the  (+) strand only display of variation context. I think we need to try and display the sequence in the correct orientation or users will continue to be confused and report that WormBase has incorrect data.

```
To whom it may concern,

I have been working with the ju476 allele of dlk-1 and I wanted to let you know that the insertion is annotated incorrectly in WormBase.

The insertion is shown to be in the "-" position of CA-CC, when it is actually one basepair over at CAC-C.

Also, the insertion is actually the reverse complement of the annotation in WormBase (i.e. instead of TAACG, it should be CGTTA).

Here is what is annotated on WormBase: CGCC CGTCCGACCAGCGACCGTCATAACGCCGTTACGGTACAGAGTTTGGGAGT.

It should be:                                              CGCC CGTCCGACCAGCGACCGTCACCGTTACGTTACGGTACAGAGTTTGGGAGT.
 

Best,
Emily Atlas

**Comment by Paul-Davis** [comment]
@sibyl229 could we revisit the display issue as I'm sure WormBase used to display in the correct orientation to maintain context.

Thanks

Paul

**Comment by tharris** [comment]
It's worth first understanding the whole history here, which includes dealing with such cases as having coordinates but no sequence, having flanks but no coordinates, having flanks and variation sequence reported on different strands, etc. At some point, I believe all variations were recorded in the database as being on the +ve strand regardless of the relevant transcriptional unit. At any rate, the code tries to deal with these many variations on this theme. I'm not sure if things changed with the display (or more likely, the underlying processing) with conversion to d2c.

**Comment by Paul-Davis** [comment]
Sure, we maybe need to evaluate the data, but I believe that they have always been curated in the context of the effected gene where that has been reported in the submission or for large scale variants captured on the strand that was supplied. I think the web code has changed as I don't recall users reporting the same issue in the past? I think there was code to display in the orientation that the variant had been curated and surely this should be the default?

**Comment by Paul-Davis** [comment]
sorry..need to edit the above before taking further comment.

**Comment by Paul-Davis** [comment]
I'm not sure what d2c is @tharris :)

**Comment by tharris** [comment]
There is a lot happening here that needs to be unpacked.

1. Some things seem to have gone missing. Conceptual translations, for example, are gone for some variations (see m2 and e105 for examples)

2. The headers on "Protein change" sequence entries (when displayed) have a duplicated "> wildtype".

3. The "Sequence" field -- formerly called "Context" is just a snippet of sequence with a random flank of 250 bp. It's not intended to show the effect on the transcriptional unit which is shown in conceptual translations (when they are displayed). I believe the problem with the extracted sequence is that it doesn't handle ambiguous bases at all.  Sure, we can place this fragment on whatever strand you wish it has nothing to do with a transcriptional unit. And for this one case with an ambiguous base, it might break in other cases where flanks are flipped or reported on different strands. It's happened, which is why there is so much code dealing with these.

**Comment by sibyl229** [comment]
@a8wright inviting you to this discussion as you are working on converting the Molecular details widget. It might be good to note some of the edge cases mentioned here.

@Paul-Davis @tharris I think it would be nice to report the sequence change on the strand that the transcriptional unit is on, if we can unambiguously and correctly identify it. A few questions:

- Once the data issue is addressed, will flanks and variation sequence always reported on the same strand as the transcriptional unit?  If not, what would be a reliable way to identify strand of the transcriptional unit and strand of variation.

- Are there cases where the strand to report variation is ambiguous due to biology (say multiple transcriptional units on opposite strands)? If so, what would be a good way to identify those?

**Comment by a8wright** [comment]
@Paul-Davis @tharris @sibyl229 This molecular details widget on the variation page is still running off of AceDB. I actually have been working on this widget today. There is a lot of logic with this widget and will take me a little while. I know of at least one more issue with this widget that has come up lately. Anything that you two would like change should be added to the list so please let me know.

**Comment by tharris** [comment]
@sibyl229 most variations at WB aren't associated with a gene from a functional perspective. These include polymorphisms and variations identified via high throughput screens. In the past, variation flanks have NOT always been reported on the same transcriptional strand as a gene they might perturb. It's always pertinent to test the orientation of flanks by looking at alignment with genomic sequence, especially if creating downstream conceptual translations with substituted variations.

**Comment by sibyl229** [comment]
@tharris Thanks. That makes sense.

**Comment by Paul-Davis** [comment]
@tharris yeah you are right, I've also found variants that have been curated on one strand and the gene they effect is on the opposite strand. I guess as long as the curated flanks and the sequence that is inserted is reported on the same strand as the flanks that are used to map the variant then the web code could do clever things to display the data on the same strand as the overlapping gene where reverse complementing the sequence and the stored text string to keep everything on the same strand.

**Comment by Paul-Davis** [comment]
@tharris yeah you are right, I've also found variants that have been curated on one strand and the gene they effect is on the opposite strand. 

I guess as long as the curated flanks and the sequence that is inserted is reported on the same strand as the flanks that are used to map the variant then the web code could do clever things to display the data on the same strand as the overlapping gene where reverse complementing the sequence and the stored text string to keep everything on the same strand.

or 

We display the feature on the strand that it has been curated on and forget trying to reverse complement etc.

Then only consider the consequence of the insertion/deletion in the widgets/pages where this is needed?

Paul

**Comment by khowe** [comment]
As a matter of policy, we standardize on the a forward strand representation when importing large variation data sets (e.g. from the million mutation project).

Since this accounts for the vast majority of data points (if not the most important ones), you will find that the strands are different approx half the time (since half of the genes are on the reverse strand).

**Comment by sibyl229** [comment]
It's totally doable to compute reverse complementary whenever necessary. But I'm not sure how to derive the correct strand. Shall I use the Gene or Predicted_CDS tag? Are there edge cases where the tags are not populated or there are contradictory multiple entries?

**Comment by khowe** [comment]
@Paul-Davis The Emily Atlas issue above is interesting:

- The flanks are stored in the db on the forward strand
- The variation feature is mapped to (and stored on) the forward strand
- The insertion sequence is w.r.t. the forward strand

Now, whether you agree with it or not, the Molecular details widget is reporting (or attempting to report) context and ref/alt alleles on the forward strand. However, in this case, the insertion sequence is incorrectly being reverse complimented for display. @sibyl229 it would be useful to look into why the display code is doing this. 

For what its worth, I think it's completely fair game for the widget to normalise on the forward strand for the sequence context. If you try to use the strand of the overlapping transcripts, then as @tharris points out, most of the time there will not be any. And there will also be pathological cases of overlap with two genes on different strand (i.e. anti-sense genes). 

Yes, strand issues cause great confusion to users (and me). But personally I find standardising on the forward strand reduces the confusion. It's also how the rest of bioinformatics does it (see VCF format)

**Comment by epaule** [comment]
@Paul-Davis do you want to keep that ticket open?
